styx-common/src/main/java/com/spotify/styx/publisher/Publisher.java

```diff
@@ -22,12 +22,13 @@
 
 import com.spotify.styx.model.ExecutionDescription;
 import com.spotify.styx.model.WorkflowInstance;
+import java.io.Closeable;
 import java.io.IOException;
 
 /**
  * An interface for publishing events.
  */
-public interface Publisher {
+public interface Publisher extends Closeable {
```

Comment:
For open source part, `Closeable` feels less surprising. `AutoCloseable` only gives a bit benefit to internal stuff.